app/_src/policies/hostnamegenerator.md

```diff
@@ -0,0 +1,72 @@
+---
+title: HostnameGenerator
+---
+
+{% warning %}
+This resource is experimental!
+{% endwarning %}
+
+A `HostnameGenerator` provides
+
+- a template to generate hostnames from properties of `MeshServices` and `MeshExternalServices`
+- a selector that defines for which `MeshServices` and `MeshExternalServices` this generator runs
+
+{% policy_yaml hg-all %}
+```yaml
+type: HostnameGenerator
+name: all
+spec:
+  selector:
+    meshService:
+      matchLabels:
+        k8s.kuma.io/namespace: kuma-demo
+  template: "{{ .DisplayName }}.{{ .Namespace }}.mesh"
+```
+{% endpolicy_yaml %}
+
+## Template
+
+A template is a [golang text template](https://pkg.go.dev/text/template).
+It is run with the name, namespace and mesh as input and function `label` to retrieve labels of the `MeshService` or `MeshExternalService`.
+
+For exampe, given:
+
+```yaml
+kind: MeshService
+metadata:
+  name: redis
+  namespace: kuma-demo
+  labels:
+    kuma.io/mesh: products
+    team: backend
+    k8s.kuma.io/service-name: redis
+    k8s.kuma.io/namespace: kuma-demo
+```
+
+and
+
+```
+  template: "{{ .DisplayName }}.{{ .Namespace }}.{{ .Mesh }}.{{ label "team" }}.mesh"
+```
+
+the generated hostname would be:
+
+```
+redis.kuma-demo.products.backend.mesh
+```
+
+Currently the generated hostname points to the first VIP known for the
+`MeshService`.
+
+## Status
+
+Every generated hostname is recorded on the `MeshService` status in `addresses`:
```

Comment:
That's also the case for MeshExternalService right?

Also what happens when an hostname is delivered twice?

Reply:
It's only recorded once